lib/oneview-sdk/image-streamer/resource/api300/golden_image.rb

```diff
@@ -10,13 +10,123 @@
 # language governing permissions and limitations under the License.
 
 require_relative 'resource'
+require 'net/http/post/multipart'
 
 module OneviewSDK
   module ImageStreamer
     module API300
       # Golden Image resource implementation for Image Streamer
       class GoldenImage < Resource
         BASE_URI = '/rest/golden-images'.freeze
+        READ_TIMEOUT = 300 # in seconds, 5 minutes
+        ACCEPTED_FORMATS = %w(.zip .ZIP).freeze # Supported upload extensions
+
+        # Create a resource object, associate it with a client, and set its properties.
+        # @param [OneviewSDK::ImageStreamer::Client] client The client object for the Image Streamer appliance
+        # @param [Hash] params The options for this resource (key-value pairs)
+        # @param [Integer] api_ver The api version to use when interracting with this resource.
+        def initialize(client, params = {}, api_ver = nil)
+          super
+          # Default values:
+          @data['type'] ||= 'GoldenImage'
+        end
+
+        # Download the details of the golden image capture logs which has been archived based on the specific attribute ID.
+        # @param [String] file_path
+        # @return [True] When was saved successfully
+        def get_details_archive(file_path)
```

Comment:
Minor thing here, but I'd probably recommend it being named `download_details_archive`. Either way is OK, but perhaps this way it's more clear that it's saving it to a file?